src/plugins/preformat/index.js

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import onKeyDown from './handlers'
+import renderBlock from './render'
+import make_schema from './schema'
+
+const ALLOWED_INLINES = ['code', 'docref', 'link', 'term', 'xref']
+
+/**
+ * @param {string[]} options.inlines - List of inline types which may appear
+ *                                     inside a preformat fragment.
+ * @param {string[]} options.marks - List of mark types which may appear inside
+ *                                   a preformat.
+ */
+export default function Preformat(options={}) {
+    const {
+        inlines = [],
+        marks = [],
+    } = options
+
+    const schema = make_schema({
+        inlines: inlines.concat(ALLOWED_INLINES),
```

Comment:
You shouldn't concatenate type arrays, as you're likely to get duplicate entries.